fusillade-api.yml

```diff
@@ -4,164 +4,465 @@ info:
-        401:
-          description: the principal is not authorized
+        - $ref: '#/components/parameters/user_id'
+      responses:
+        200:
+          description: Information about the user is returned
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/User'
+        404:
+          description: Resource not found.
         default:
           description: Unexpected error
-  /users:
     put:
-      summary: Create or update a user and attached policy.
+      summary: Create a new user
       security:
-        - openId:
-            - https://www.googleapis.com/auth/userinfo.email
+        - bearerAuth: []
       parameters:
-        - name: json_request_body
-          in: body
-          required: true
-          schema:
-            type: object
-            properties:
-              policy:
-                description: IAM policy to attach to the user, if any
-                type: object
+        - $ref: '#/components/parameters/user_id'
+      requestBody:
+        content:
+          application/json:
+            schema:
+              type: object
+              properties:
+                groups:
+                  type: array
+                  items:
+                    type: string
+                roles:
+                  type: array
+                  items:
+                    type: string
+                policy:
+                  type: string
       responses:
-        201:
-          description: OK
-          schema:
-            type: object
+        200:
+          description: User was created
+        401:
+          description: The user is not authorized because the object has already been created and the user does not have permission to modify, or user does not have permission to create objects.
         default:
           description: Unexpected error
-  /users/{user_id}:
+  /users/{user_id}/groups:
     get:
-      summary: "Retrieve information about a user."
+      summary: Retrieve groups a user is in.
       security:
-        - Authorization:
-            - null
-        - openId:
-            - https://www.googleapis.com/auth/userinfo.email
+        - bearerAuth: []
       parameters:
-        - name: user_id
-          in: path
-          description: User ID (email)
-          required: true
-          type: string
+        - $ref: '#/components/parameters/user_id'
       responses:
         200:
           description: OK
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  - group:
+                      type: array
+                      items:
+                        type: string
+        404:
+          description: Resource not found.
         default:
           description: Unexpected error
-  /groups:
     put:
-      summary: Create or update a group and attached policy.
+      summary: Add user to groups
```

Comment:
This was changed to `add` but the params below still allow specifying `add` or `delete` (`- $ref: '#/components/parameters/action'`).

Reply:
resolved